components/bl-treeSelect-component/src/index.js

```diff
@@ -89,6 +85,20 @@ function useNodeActions(nodes, eventHandlers, setSelectedNodeKey, selectionMode)
 
   const [expandedKeys, setExpandedKeys] = useState({});
 
+  const keysMap = useMemo(() => {
+    const keys = {};
+
+    if (!nodes?.length) {
+      return keys;
+    }
+
+    for (const childNode of nodes) {
+      updateKeysMap(childNode, keys);
+    }
+
+    return keys;
+  }, [nodes]);
```

Comment:
Could be moved into hook useKeysMap. For the reason to make component body shorter

Reply:
But it is placed not in the component body but in the custom hook.  Do you think it would be better to move it to another hook?